### PR TITLE
feat: propagate newrelic tx info by using fiber.Ctx and little refact…

### PR DESCRIPTION
**Pull request by Abdulsametileri**
In order to create segments we need to pass created new relic transaction using fiber.Ctx to the chain.

### CODE REVIEW

File: fibernewrelic/fiber.go [hunk 1]
```diff
@@ -66,33 +65,26 @@ func New(cfg Config) fiber.Handler {
 	}
 
 	return func(c *fiber.Ctx) error {
-		txn := app.StartTransaction("")
+		txn := app.StartTransaction(createTransactionName(c))
 		defer txn.End()
 
-		handlerErr := c.Next()
-
-		var (
```
Comment (Abdulsametileri) [inline fibernewrelic/fiber.go:74]: No need to do any copy for more clear implementation, I recommend to review it as `View File` section 😅

**Comment by ReneWerner87** [comment]
@mstrYoda can you review

**Comment by mstrYoda** [comment]
It seems ok to me 👍  I talked with the author, he will add a benchmark test to see if allocation count increased or not 👍

**Comment by Abdulsametileri** [comment]
Here the benchmark results

```go
go test -bench='.' -benchtime='3s' -benchmem                                                                    ✔  9s  10:30:52 
goos: darwin
goarch: arm64
pkg: github.com/gofiber/contrib/fibernewrelic
BenchmarkNew-10           297830             10180 ns/op            7770 B/op         45 allocs/op
BenchmarkOld-10           362887              9865 ns/op            7695 B/op         45 allocs/op
PASS
ok      github.com/gofiber/contrib/fibernewrelic        10.122s
```

allocation is OK!

ns/op increased because we added newrelic transaction `c.SetUserContext(newrelic.NewContext(c.UserContext(), txn))`

**Review by mstrYoda — APPROVED** [review]

**Comment by mstrYoda** [comment]
Thank you for your contribution @Abdulsametileri 🙏

**Comment by Abdulsametileri** [comment]
Hello @ReneWerner87, is it possible to create a new release for the fibernewrelic package? 🤔

**Comment by ReneWerner87** [comment]
https://github.com/gofiber/contrib/releases/tag/fibernewrelic%2Fv1.2.0

**Pull request merged**